### ISSUE TITLE
Unused EMS impacts simulation results

### ISSUE THREAD

**Issue by lymereJ** [NREL/EnergyPlus#8164]
Issue overview
--------------
We observe differences in results produced by two IDFs that are virtually identical. The only difference being that one of them includes two EMS programs. Only one of these EMS program uses an actuator. The actuator changes the output of a curve but the curve isn’t used by any objects in the IDF file. For that reason, we would expect the results to be identical, however, they are not. The simulation results show differences in cooling energy use (3% in NYC). We cannot seem to explain these differences. We noticed this in EnergyPlus 8.9, but we observed the same results after upgrading to 9.1.

### Details
Some additional details for this issue (if relevant):
 - Platform:  Windows 10
 - Version of EnergyPlus: 8.9, 9.1

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: sent to @mjwitte by email.
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by jmarrec** [comment]
Could you share the file (or better, a MCVE - minimal, complete, verifiable example) please? privately if need be.

**Comment by mjwitte** [comment]
Defect files added to EnergyPlusDevSupport\DefectFiles

**Comment by jmarrec** [comment]
![image](https://user-images.githubusercontent.com/5479063/88408702-3853d380-cdd4-11ea-9a37-818762e1fcc7.png)

**Comment by jmarrec** [comment]
I've narrowed it down to a MCVE (EMS-wise) and removed everything (the Test Curve, and all EMS objects) But this:

```
  EnergyManagementSystem:ProgramCallingManager,
    xxxBoilerCurveManager,      !- Name
    InsideHVACSystemIterationLoop,  !- EnergyPlus Model Calling Point
    xxxSetBoilerCurve;          !- Program Name 1


  EnergyManagementSystem:Program,
    xxxSetBoilerCurve,          !- Name
    ;                                       !- Program Line 1
```

This only suffices to produce the difference.

**Comment by jmarrec** [comment]
Changing the calling point to `AfterPredictorAfterHVACManagers` removes the difference

**Comment by jmarrec** [comment]
Will need to check what happens around 2005-01-07 06:00:00

**Comment by jmarrec** [comment]
Ok, so after a deep investigation, this happens if you have any EMS program (whether it does something or not) that has the calling point `InsideHVACSystemIterationLoop` because it will increase the number of iterations significantly, because AirLoopHVACs will simulate more often.

Here is the difference with the 5ZoneAirCooled MCVE I created, for a simulation from 1/16 and 1/19:

![image](https://user-images.githubusercontent.com/5479063/94256291-e3dee880-ff29-11ea-92f4-004c2bffc4d8.png)


The main part is that `HVACManageIteration` will always be at least 3.

https://github.com/NREL/EnergyPlus/blob/b1c7a5b4634ed662ef93c23b195ba084a09df3c6/src/EnergyPlus/HVACManager.cc#L933-L936

![image](https://user-images.githubusercontent.com/5479063/94256172-a8dcb500-ff29-11ea-8df3-3a9be97ccb12.png)


I am voting to close this issue as won't fix, as it would introduce way too much overhead to check whether actuators are actually:

1) Actuated
2) Actuating a property of an object that is actually used in the simulation

**Comment by mjwitte** [comment]
I agree that it would be too difficult to determine whether the EMS program actuated anything that is actually being used.
But a couple of extra iterations should not be moving the answer significantly.

**Comment by jmarrec** [comment]
The 5ZoneAirCooled.idf has a lot of unmet demand rate, increasing the number of iterations reduces it significantly.

df_noems[('HOT WATER LOOP', 'Plant Supply Side Unmet Demand Rate', 'W')].sum()

401526.3302085112

df_ems[('HOT WATER LOOP', 'Plant Supply Side Unmet Demand Rate', 'W')].sum()

15935.358127248337

**Comment by jmarrec** [comment]
Blue is the NO EMS

![image](https://user-images.githubusercontent.com/5479063/94280650-ece1b100-ff4d-11ea-994a-4b14ab5f6a4a.png)



It's happening at very low PLR. The boiler has a min PLR of 0 though:

```
  Boiler:HotWater,
    Central Boiler,          !- Name
    NaturalGas,              !- Fuel Type
    autosize,                !- Nominal Capacity {W}
    0.8,                     !- Nominal Thermal Efficiency
    LeavingBoiler,           !- Efficiency Curve Temperature Evaluation Variable
    BoilerEfficiency,        !- Normalized Boiler Efficiency Curve Name
    autosize,                !- Design Water Flow Rate {m3/s}
    0.0,                     !- Minimum Part Load Ratio
    1.2,                     !- Maximum Part Load Ratio
    1.0,                     !- Optimum Part Load Ratio
    Central Boiler Inlet Node,  !- Boiler Water Inlet Node Name
    Central Boiler Outlet Node,  !- Boiler Water Outlet Node Name
    100.,                    !- Water Outlet Upper Temperature Limit {C}
    LeavingSetpointModulated;!- Boiler Flow Mode
```

**Comment by jmarrec** [comment]
Something in happening the bypass. Here is a diff (multiply by 100 to get percents) of all variables I requested.

![image](https://user-images.githubusercontent.com/5479063/94282181-cfade200-ff4f-11ea-8ce3-4ff0efbe9e69.png)

Here are the bypass flowrates:

![image](https://user-images.githubusercontent.com/5479063/94283293-1f40dd80-ff51-11ea-9b65-9bfbfcb3ec71.png)


![image](https://user-images.githubusercontent.com/5479063/94283381-37b0f800-ff51-11ea-919a-c9977b494ebe.png)

**Comment by jmarrec** [comment]
# Switching the `Boiler:HotWater` from `LeavingSetpointModulated` to `Modulated` makes the difference go away.


## LeavingSetpointModulated

![image](https://user-images.githubusercontent.com/5479063/94284864-439db980-ff53-11ea-96a4-fa76ae6b4cd7.png)


## NotModulated

![image](https://user-images.githubusercontent.com/5479063/94284621-ee61a800-ff52-11ea-837e-39bf80cd4c5c.png)

**Comment by brianlball** [comment]
@jmarrec @lymereJ was this resolved?

**Comment by lymereJ** [comment]
I don't think so?
